### PR TITLE
Add JSON Functions

### PR DESCRIPTION
**Pull request by jpb**
- add `!$toJson` and `!$parseJson`
- add `!$toYaml` as an alias for `!$string`
- accept non literal values as input to `!$parseJson` and `!$string`/`!$toYaml`

### CODE REVIEW

File: docs/yaml-preprocessing.md [hunk 1]
```diff
@@ -456,6 +456,7 @@ things: !$escape
 ```
 
 * `!$string` convert data to a YAML string
+`!$toYaml` is an alias of `!$string`
```
Comment (tavisrudd) [inline docs/yaml-preprocessing.md:459]: How about `toYamlString`?

Comment (tavisrudd) [reply docs/yaml-preprocessing.md:459]: We should also documented the alternate `{{toyaml xxx}}` version and `{{tojson xxx}}', `{{tojsonPretty xxx}}`. Speaking of them, should the names of each version have the same casing?

Comment (jpb) [reply docs/yaml-preprocessing.md:459]: > How about `toYamlString`?

Sure 👍 

> We should also document the alternate `{{toyaml xxx}}` version and `{{tojson xxx}}`, `{{tojsonPretty xxx}}`. Speaking of them, should the names of each version have the same casing?

Those were documented in #251. 

I agree they should be consistent – it seem like adding `camelCase` aliases to those that don't match the format (in another PR) would be the best way forward – what do you think?

Comment (tavisrudd) [reply docs/yaml-preprocessing.md]: 👍

Comment (jpb) [reply docs/yaml-preprocessing.md]: Done in #253.

**Pull request merged**